services/cd-service/pkg/repository/repository.go

```diff
@@ -947,6 +949,15 @@ func (r *repository) ApplyTransformers(ctx context.Context, transaction *sql.Tx,
 		return nil, &TransformerBatchApplyError{TransformerError: fmt.Errorf("%s: %w", "failure in afterTransform", err), Index: -1}
 	}
 
+	gitMutexLock.Lock()
+	defer gitMutexLock.Unlock()
+	if state.DBHandler.ShouldUseOtherTables() {
+		var err error
+		state, err = r.StateAt(nil)
```

Comment:
and why do we need `StateAt` when we didn't need it before?

Reply:
Because when creating releases in parallel we need to make sure that we're at the latest commit. This will be removed in the followup story.